### ISSUE TITLE
Resource filename conventions not working, when assembly name and default namespace differ

### ISSUE THREAD

**Issue by TsengSR** [aspnet/Localization#398]
I spent the better half of yesterday trying to get the locations work, after I moved my controllers to a different assembly and todays morning to create a small reproducible project to demonstrate it. 

With the first setup I did, it worked to my surprise so I had to spend a good hour trying to bring it to the constellation I have it in current project. 

Finally I pinned it down that neither of the (from my point of view) expected filenames for resource files worked, when assembly name and default namespace did differ. 

Here is a link to a demonstration project: https://github.com/TsengSR/LocalizationExample

When the namespace of the class library is `LocalizationExample.Mvc` and the default namespace `LocalizationExample`, then neither of the resources are loaded.

`http://localhost:62979/api/mvc` will just return the untranslated string "Hello Mvc". 

I've tried 

* SharedMvcResources.de-DE.resx
* LocalizationExample.SharedMvcResources.de-DE.resx
* LocalizationExample.Mvc.SharedMvcResources.de-DE.resx
* LocalizationExample.Mvc.LocalizationExample.SharedMvcResources.de-DE.resx
* LocalizationExample.LocalizationExample.SharedMvcResources.de-DE.resx

Basically all kind of permutations of assembly name, namespace and class name. 

When changing both, the assembly name and default namespace to `LocalizationExample.Mvc` or `LocalizationExample` then it works (getting `LocalizationExample.SharedMvcResources.de-DE.resx` and `SharedMvcResources.de-DE.resx` respectively). 

As for the reasoning/use case: 
I often keep the default namespace different from assembly name, due to separation of concerns and not wanting that every newly created .cs file gets the assembly name in the namespace.

**Comment by hishamco** [comment]
If I'm not wrong this a duplicate for issue #340

**Comment by mkArtakMSFT** [comment]
Closing this as a dupe of https://github.com/aspnet/Localization/issues/340. Will use the original issue for tracking.
